pkg/image/anaconda_net_installer.go

```diff
@@ -0,0 +1,151 @@
+	EnabledAnacondaModules  []string
+	DisabledAnacondaModules []string
+
+	AdditionalDracutModules []string
+	AdditionalDrivers       []string
+
+	// Uses the old, deprecated, Anaconda config option "kickstart-modules".
+	// Only for RHEL 8.
+	UseLegacyAnacondaConfig bool
+}
+
+func NewAnacondaNetInstaller() *AnacondaNetInstaller {
+	return &AnacondaNetInstaller{
+		Base: NewBase("netinst"),
+	}
+}
+
+func (img *AnacondaNetInstaller) InstantiateManifest(m *manifest.Manifest,
+	repos []rpmmd.RepoConfig,
+	runner runner.Runner,
+	rng *rand.Rand) (*artifact.Artifact, error) {
+	buildPipeline := addBuildBootstrapPipelines(m, runner, repos, nil)
+	buildPipeline.Checkpoint()
+
+	installerType := manifest.AnacondaInstallerTypeNetinst
+	anacondaPipeline := manifest.NewAnacondaInstaller(
+		installerType,
+		buildPipeline,
+		img.Platform,
+		repos,
+		"kernel",
+		img.Product,
+		img.OSVersion,
+		img.Preview,
+	)
+
+	anacondaPipeline.ExtraPackages = img.ExtraBasePackages.Include
+	anacondaPipeline.ExcludePackages = img.ExtraBasePackages.Exclude
+	anacondaPipeline.ExtraRepos = img.ExtraBasePackages.Repositories
+	anacondaPipeline.Variant = img.Variant
+	anacondaPipeline.Biosdevname = (img.Platform.GetArch() == arch.ARCH_X86_64)
+
+	anacondaPipeline.UseLegacyAnacondaConfig = img.UseLegacyAnacondaConfig
+	anacondaPipeline.EnabledAnacondaModules = img.EnabledAnacondaModules
+	if img.OSCustomizations.FIPS {
+		anacondaPipeline.EnabledAnacondaModules = append(
+			anacondaPipeline.EnabledAnacondaModules,
+			anaconda.ModuleSecurity,
+		)
+	}
+	anacondaPipeline.DisabledAnacondaModules = img.DisabledAnacondaModules
+	anacondaPipeline.AdditionalDracutModules = img.AdditionalDracutModules
+	anacondaPipeline.AdditionalDrivers = img.AdditionalDrivers
+	anacondaPipeline.Locale = img.OSCustomizations.Language
+
+	anacondaPipeline.Checkpoint()
+
+	var rootfsImagePipeline *manifest.ISORootfsImg
+	switch img.RootfsType {
+	case manifest.SquashfsExt4Rootfs:
+		rootfsImagePipeline = manifest.NewISORootfsImg(buildPipeline, anacondaPipeline)
+		rootfsImagePipeline.Size = 5 * datasizes.GibiByte
+	default:
+	}
+
+	bootTreePipeline := manifest.NewEFIBootTree(buildPipeline, img.Product, img.OSVersion)
+	bootTreePipeline.Platform = img.Platform
+	bootTreePipeline.UEFIVendor = img.Platform.GetUEFIVendor()
+	bootTreePipeline.ISOLabel = img.ISOLabel
+
+	kernelOpts := []string{fmt.Sprintf("inst.stage2=hd:LABEL=%s", img.ISOLabel)}
+	if img.OSCustomizations.FIPS {
+		kernelOpts = append(kernelOpts, "fips=1")
+	}
+	kernelOpts = append(kernelOpts, img.AdditionalKernelOpts...)
+	bootTreePipeline.KernelOpts = kernelOpts
+
+	// Payload type uses an OS Pipeline for the payload
+	var osPipeline *manifest.OS
+	if anacondaPipeline.Type == manifest.AnacondaInstallerTypePayload {
+		osPipeline = manifest.NewOS(buildPipeline, img.Platform, repos)
+		osPipeline.OSCustomizations = img.OSCustomizations
+		osPipeline.Environment = img.Environment
+		osPipeline.Workload = img.Workload
+	}
```

Comment:
This is technically dead code, since `anacondaPipeline.Type` is always set to `manifest.AnacondaInstallerTypeNetinst` above.